drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1706,7 +1706,7 @@ static void eth0_iface_init(struct net_if *iface)
 
 	i = 0;
 	for (j = NET_PRIORITY_NC; j >= 0; --j) {
-		if (j > 7) {
+		if (i > 7) {
```

Comment:
Ok, still, how about: a) using ARRAY_SIZE(GMAC_ST2RPQ) here; b) moving the check to right before the `i` is used, i.e. after the `if (priority2queue(j) == 0) continue;`.

Reply:
@pfalcon no problem, it makes perfect sense to do this!

I've put the _refactoring_ changes in a separate commit so it doesn't look like a refactor change changes the functionality, is that ok?

Reply:
> I've put the refactoring changes in a separate commit so it doesn't look like a refactor change changes the functionality, is that ok?

I'm mot sure, I'd squash it, and may imagine you may be asked to  by other folks. But ok by me, thanks!

Reply:
Ok, I thought it would be better not to mix fixes and refactoring, but if you anticipate comments about this, then I have nothing against it either. It is a small refactor after all.

Just pushed a new revision with the changes squashed. Thanks again!